src/main/java/de/tum/in/www1/artemis/web/rest/dto/competency/KnowledgeAreaDTO.java

```diff
@@ -10,7 +13,8 @@
  * DTO containing {@link KnowledgeArea} data. It only contains the id of its parent.
  */
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
-public record KnowledgeAreaDTO(long id, String title, String shortTitle, String description, Long parentId, List<KnowledgeAreaDTO> children,
+@JsonIgnoreProperties(ignoreUnknown = true)
```

Comment:
@JohannesWt said it would be a good use here - Especially since we have Objects in the client that are KnowledgeAreaDTOs with extra information and we might not want to always map them back to DTOs. (This is not happening right now but might be used in the future)

Reply:
The thing is, for request, this would only be necessary if right now the client would receive an exception if they provide a property that is not part of the DTO. If that's not the case, we don't need the annotation for requests (though this is exactly the case that I would then configure globally, as we don't want that in general to happen).

For the response this annotation is irrelevant as we don't map the data to the client DTO, but we serialise it in a JSON object. What happens with that JSON object in the client is only the client's concern.

So my only question would be, do we get exceptions when the client calls the server with more properties? If not, I don't see the annotation doing anything.

Reply:
I double-checked this, and you are right: Any extra fields sent by the client are ignored. I'll remove the annotations.